src/artwork.c

```diff
@@ -1744,7 +1744,11 @@ source_item_ownpl_get(struct artwork_ctx *ctx)
 	continue;
 
       ctx->dbmfi->path = dbpli.path;
-      format = source_item_own_get(ctx);
+      if (dbpli.artwork_url)
+        format = artwork_get(ctx->evbuf, dbpli.artwork_url, NULL, ctx->max_w, ctx->max_h, false, ctx->data_kind);
+
+      if (format == ART_E_NONE)
+        format = source_item_own_get(ctx);
```

Comment:
source_item_own_get() isn't really supposed to be called with URL's. Before RSS feeds were added playlist paths were never URL's, so it couldn't happen.. I suggest you include a guard here so that the function is not called with remote paths.

Reply:
As `source_item_own_get()` isn't meant for remote paths I'm guarding against this via
```
if (dbpli.path && dbpli.path[0] == '/')
```
and saving some cycles instead of `strncmp` against `http://` or `https://`